app.py

```diff
@@ -143,47 +150,66 @@ def log_out():
 
 
 @app.route("/admin/libraries")
+@auth_admin_only
```

Comment:
I'd suggest calling this something more descriptive like: `require_admin_authentication`